src/v4/api/extendobservable.ts

```diff
@@ -64,7 +64,7 @@ export function extendObservable<A extends Object, B extends Object>(
     asObservableObject(target, options.name, defaultDecorator.enhancer) // make sure object is observable, even without initial props
     startBatch()
     try {
-        for (let key in properties) {
```

Comment:
`for ... in ...` contains enumerable properties that the `properties` inherits from its prototype chain. 
But `Object.getOwnPropertyNames` only contains enumerable properties of `properties` itself. 
@mweststrate I'm not sure the difference matters.

Reply:
@MoonBall good one, no for extendObservable that should be match. However, If I remember correctly the downlevel compilation of `for of` loops is problematic with MobX 4, which might run in environments without iterators. Would you mind changing it to a `.forEach` or class `for(;;)` loop?

Reply:
I fix `for of` by using `for(;;)`.